gguf-py/gguf/metadata.py

```diff
@@ -0,0 +1,408 @@
+
+        metadata.size_label                = metadata_override.get(Keys.General.SIZE_LABEL,                metadata.size_label)
+        metadata.license_name              = metadata_override.get(Keys.General.LICENSE_NAME,              metadata.license_name)
+        metadata.license_link              = metadata_override.get(Keys.General.LICENSE_LINK,              metadata.license_link)
+
+        metadata.url                       = metadata_override.get(Keys.General.URL,                       metadata.url)
+        metadata.doi                       = metadata_override.get(Keys.General.DOI,                       metadata.doi)
+        metadata.uuid                      = metadata_override.get(Keys.General.UUID,                      metadata.uuid)
+        metadata.repo_url                  = metadata_override.get(Keys.General.REPO_URL,                  metadata.repo_url)
+
+        metadata.source_url                = metadata_override.get(Keys.General.SOURCE_URL,                metadata.source_url)
+        metadata.source_doi                = metadata_override.get(Keys.General.SOURCE_DOI,                metadata.source_doi)
+        metadata.source_uuid               = metadata_override.get(Keys.General.SOURCE_UUID,               metadata.source_uuid)
+        metadata.source_repo_url           = metadata_override.get(Keys.General.SOURCE_REPO_URL,           metadata.source_repo_url)
+
+        # Base Models is received here as an array of models
+        metadata.base_models               = metadata_override.get("general.base_models",                  metadata.base_models)
+
+        metadata.tags                      = metadata_override.get(Keys.General.TAGS,                      metadata.tags)
+        metadata.languages                 = metadata_override.get(Keys.General.LANGUAGES,                 metadata.languages)
+        metadata.datasets                  = metadata_override.get(Keys.General.DATASETS,                  metadata.datasets)
+
+        # Direct Metadata Override (via direct cli argument)
+        if model_name is not None:
+            metadata.name = model_name
+
+        return metadata
+
+    @staticmethod
+    def load_metadata_override(metadata_override_path: Optional[Path] = None) -> dict[str, Any]:
+        if metadata_override_path is None or not metadata_override_path.exists():
+            return {}
+
+        with open(metadata_override_path, "r", encoding="utf-8") as f:
+            return json.load(f)
+
+    @staticmethod
+    def load_model_card(model_path: Optional[Path] = None) -> dict[str, Any]:
+        if model_path is None or not model_path.exists():
+            return {}
+
+        model_card_path = model_path / "README.md"
+
+        if not model_card_path.exists():
+            return {}
+
+        with open(model_card_path, "r", encoding="utf-8") as f:
+            return cast("dict[str, Any]", frontmatter.load(f))
```

Comment:
```suggestion
            return frontmatter.load(f).to_dict()
```

`frontmatter.load()` returns a `Post` which has a `to_dict` method.